Incremental Store/EncryptedStore.m

```diff
@@ -965,12 +966,21 @@ - (BOOL)loadMetadata:(NSError **)error {
                     if ([newModel isConfiguration:nil compatibleWithStoreMetadata:metadata]){
                         return YES;
                     }
-                    
-                    // load the old model:
-                    NSMutableArray *bundles = [NSMutableArray array];
-                    NSBundle *bundle = self.fileManager.configuration.bundle;
-                    [bundles addObject:bundle];
-                    NSManagedObjectModel *oldModel = [NSManagedObjectModel mergedModelFromBundles:bundles
+
+                    // load the old model
+                    NSBundle *modelBundle;
+                    NSString *bundleId = [options objectForKey:EncryptedStoreBundleId];
+                    if (bundleId.length > 0)
+                    {
+                        modelBundle = [NSBundle bundleWithIdentifier:bundleId];
+                    }
+
+                    if (modelBundle == nil)
+                    {
+                        modelBundle = [NSBundle mainBundle];
```

Comment:
Here the main bundle is fetched and then later added to an array on line 983. However according to Apple documentation on the [mainBundle method](https://developer.apple.com/documentation/foundation/nsbundle/1410786-mainbundle?language=objc) the method can return nil. Therefore an error will occur when attempting to add nil to an array. 

> This method may return a valid bundle object even for unbundled apps. It may also return nil if the bundle object could not be created, so always check the return value.

You could add a additional check and error code for this case if you wanted to make this code as resilient as possible. 

```
typedef NS_ENUM(NSInteger, EncryptedStoreError)
{
    EncryptedStoreErrorIncorrectPasscode = 6000,
    EncryptedStoreErrorMigrationFailed,
    EncryptedStoreErrorModelLoadFailed
};
```

```
if (modelBundle == nil)
{
    if (error)
    {
        NSDictionary * userInfo = @{EncryptedStoreErrorMessageKey : @"Main Bundle not found, cannot load model"};
        *error = [NSError errorWithDomain: EncryptedStoreErrorDomain 
                                     code: EncryptedStoreErrorModelLoadFailed 
                                 userInfo: userInfo];
    }
    return NO;
}
```

Reply:
While I agree with about handling mainBundle method this is kind of overkill since if mainBundle is nil then we are in more serious troubles than handling error correctly, 
Also getting modelBundle from mainBundle assumes that model is delivered inside app while issue which has been fixed by this PR is to migrate correctly for model delivered in framework

Reply:
Ok, perhaps it would be an extreme edge case.